server/src/main/java/com/flipkart/varadhi/deployment/LeanDeploymentVerticleDeployer.java

```diff
@@ -0,0 +1,115 @@
+            MeterRegistry meterRegistry
+    ) {
+        super(hostName, vertx, configuration, messagingStackProvider, metaStoreProvider, meterRegistry);
+        validateLeanDeploymentConstraints(configuration.getRestOptions());
+    }
+
+    @Override
+    public List<RouteDefinition> getRouteDefinitions() {
+        return List.of();
+    }
+
+    private void validateLeanDeploymentConstraints(RestOptions restOptions) {
+        String defaultOrg = restOptions.getDefaultOrg();
+        String defaultTeam = restOptions.getDefaultTeam();
+        String defaultProject = restOptions.getDefaultProject();
+
+        List<Org> orgs = orgService.getOrgs();
+
+        if (orgs.size() > 1) {
+            throw new VaradhiException("Lean deployment can not be enabled as there are more than one orgs.");
+        }
+
+        if (orgs.size() == 1 && !defaultOrg.equals(orgs.get(0).getName())) {
+            throw new VaradhiException("Lean deployment can not be enabled as org with different name is present. " +
+                    "Please delete the org or change the default org.");
+        }
+
+        if (orgs.isEmpty()) {
+            log.debug("Creating default org, team and project as no org is present.");
+
+            orgService.createOrg(new Org(defaultOrg, 0));
+            teamService.createTeam(new Team(defaultTeam, 0, defaultOrg));
+            projectService.createProject(new Project(defaultProject, 0, "", defaultTeam, defaultOrg));
+
+            log.debug("Created default org, team and project as no org is present.");
+            return;
+        }
+
+        if (isDefaultTeamPresent(defaultTeam, defaultOrg)) {
+            log.debug("validate default project");
+            validateProjectConstraints(defaultProject, defaultTeam, defaultOrg);
+        }
+    }
+
+    private void validateProjectConstraints(
+            String defaultProject,
+            String defaultTeam,
+            String defaultOrg) {
+
+        List<Project> projects = projectService.getProjects(defaultTeam, defaultOrg);
+
+        if (projects.size() > 1) {
+            throw new VaradhiException("Lean deployment can not be enabled as there are more than one projects.");
+        }
+        if (projects.size() == 1 && !defaultProject.equals(projects.get(0).getName())) {
+            throw new VaradhiException("Lean deployment can not be enabled as project with different name is present. " +
+                    "Please delete the project or change the default project.");
+        }
+        if (projects.isEmpty()) {
+            log.debug("Creating default project as no project is present.");
+
+            projectService.createProject(new Project(defaultProject, 0, "", defaultTeam, defaultOrg));
+
+            log.debug("Created default project as no project is present.");
+        }
+    }
+
+    private boolean isDefaultTeamPresent(String defaultTeam, String defaultOrg) {
```

Comment:
This is creating a team/project as well.
May be we can refactor this entire code into separate methods, each taking care of individual entity i.e. separate method for ensuring only required org, team and project exists.